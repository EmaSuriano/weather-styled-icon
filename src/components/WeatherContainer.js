import styled from 'styled-components';
import PropTypes from 'prop-types';

const WeatherContainer = styled.div`
  position: absolute;
  z-index: 2;
  top: 50%;
  left: 50%;
  width: 3.75em;
  height: 3.75em;
  margin: 0.375em 0 0 -2em;
  background: ${props => props.theme.backgroundColor};
`;

WeatherContainer.propTypes = {
  theme: PropTypes.shape({
    backgroundColor: PropTypes.string,
  }),
};

WeatherContainer.displayName = 'WeatherContainer';

export default WeatherContainer;
