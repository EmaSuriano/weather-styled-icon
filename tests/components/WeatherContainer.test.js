import React from 'react';
import WeatherContainer from '../../src/components/WeatherContainer';
import { shallow } from 'enzyme';
import 'jest-styled-components';

describe('<WeatherContainer />', () => {
  it('should should render div with theme', () => {
    const wrapper = shallow(
      <WeatherContainer theme={{ backgroundColor: 'red' }} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
