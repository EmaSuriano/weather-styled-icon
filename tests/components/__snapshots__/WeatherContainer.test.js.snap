// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherContainer /> should should render div with theme 1`] = `
.c0 {
  position: absolute;
  z-index: 2;
  top: 50%;
  left: 50%;
  width: 3.75em;
  height: 3.75em;
  margin: 0.375em 0 0 -2em;
  background: red;
}

<div
  className="c0"
/>
`;
